### PR TITLE
Fix index overflow

### LINKED ISSUE #29: Index overflow not handled.
The writer and reader indices are of the type usize which is a 32 or 64 bit unsigned integer depending on the target's address space. The edge case where the writer index has overflown pass the usize::MAX value is not covered by the internal queue logic. 
For this to be an issue on 64 bit targets more then 18,446,744,073,709,551,615 items need to have been written to the queue.

It would take more then 500 years for this to happen if a value is written to the queue every 1ns.

### PR DESCRIPTION
**Pull request by filipdulic**
Closes #29 
Adds test covering index overflow edge case.
Issue fixed using the usize::wrapping_sub and usize::wrapping_add methods.

### CODE REVIEW

File: src/channel.rs [hunk 1]
```diff
@@ -339,12 +340,63 @@ mod test {
...
+        assert_eq!(*receiver.try_recv().unwrap(), 1);
+        assert_eq!(*receiver.try_recv().unwrap(), 2);
+
+        // wi should be at usize::max_value()
+        assert_eq!(sender.wi.get(), usize::max_value());
+        // ri should be at usize::max_value() -1
+        assert_eq!(receiver.ri.get(), usize::max_value() - 1);
+
+        // broadcast 6 more items so wi is at 5
+        sender.broadcast(4).unwrap();
```
Comment (vladan) [inline src/channel.rs]:
```
for i in 4..10 {
    sender.broadcast(i);
}
```
:)

Comment (filipdulic) [reply src/channel.rs]: Thanks.
